app/assets/javascripts/discourse/app/lib/click-track.js

```diff
@@ -71,7 +71,7 @@ export function openLinkInNewTab(link) {
 }
 
 export default {
-  trackClick(e, siteSettings) {
+  trackClick(e, siteSettings, { returnPromise = false } = {}) {
```

Comment:
Without `returnPromise` tests were very janky because the redirects could happen after the test conditional was checked.